Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -103,59 +115,145 @@ protected override void OnCreate(Bundle savedInstanceState)
                         break;
                     case Resource.Id.NavItemShoppingCart:
                         var intentCart = new Intent(this, typeof(ShoppingCartActivity));
+                        intentCart.PutExtra(noTalkBackIntentKey, noTalkBack);
                         StartActivity(intentCart);
                         break;
                     case Resource.Id.NavItemShoppingList:
                         var intentList = new Intent(this, typeof(ShoppingListActivity));
+                        intentList.PutExtra(noTalkBackIntentKey, noTalkBack);
                         StartActivity(intentList);
                         break;
-                    case Resource.Id.NavItemTextToSpeech:
-                        var intentTTS = new Intent(this, typeof(TextVoicerActivity));
-                        StartActivity(intentTTS);
-                        break;
-                    case Resource.Id.NavItemSpeechRecogniser:
-                        var intentSpeech = new Intent(this, typeof(SpeechActivity));
-                        StartActivity(intentSpeech);
-                        break;
                 }
             };
         }
 
-        void RecogniseVoice(object sender, EventArgs e)
+        protected override void OnStart()
+        {
+            base.OnStart();
+            tutorialNeeded = IsTutorialNeeded();
+        }
+
+        protected override void OnStop()
         {
-            if (commandRecognizer.IsValueCreated)
+            // Get rid of TTS.
+            if (tts != null)
             {
-                speechIntent = VoiceRecognizerHelper.SetUpVoiceRecognizerIntent();
-                commandRecognizer.Value.StartListening(speechIntent);
+                tts.Stop();
```

Comment:
Don't you just hate it when you close ShopLens, but Jenna doesn't stop talking? This is the solution to that problem. Albeit if you go back to the app, it doesn't work properly.

Reply:
I wish there was a resume method for TTS, but there isn't... There's a way to pause/resume TTS by saving it's message string to a file using synthesizeToFile() and then using MediaPlayer to pause/resume the reading of the file, but I think then we wouldn't be able to use Utterance Listener: https://stackoverflow.com/questions/4970204/how-to-pause-android-speech-tts-texttospeech/4976327#4976327

Reply:
I mean not pause mid-sentence, but just restart/recreate the tts object.

Reply:
But how would we know what the tts object should say when OnResume is called? Would it just say "Hey, where were you gone? So, what would you have me do?"? That could work.